### ISSUE TITLE
Missing Value Backend not being properly populated

### ISSUE THREAD

**Issue by okennedy** [UBOdin/mimir#86]
```
> testOnly mimir.demo.SimpleDemoScript
java.sql.SQLException: [SQLITE_ERROR] SQL error or missing database (no such table: RATINGS1FINAL_1_BACKEND)during
SELECT SUBQ_RATINGS1_PID.RATINGS1_PID AS RATINGS1_PID, SUBQ_RATINGS1_PID.RATINGS1_RATING AS RATINGS1_RATING, SUBQ_RATINGS1_PID.RATINGS1_REVIEW_CT AS RATINGS1_REVIEW_CT, SUBQ_RATINGS1_PID.ROWID_MIMIR AS ROWID_MIMIR FROM (SELECT RATINGS1.PID AS RATINGS1_PID, RATINGS1.RATING AS RATINGS1_RATING, RATINGS1.REVIEW_CT AS RATINGS1_REVIEW_CT, RATINGS1.ROWID AS ROWID_MIMIR FROM RATINGS1) SUBQ_RATINGS1_PID WHERE CAST(SUBQ_RATINGS1_PID.RATINGS1_RATING AS REAL) IS NULL AND (SELECT RATINGS1FINAL_1_BACKEND.DATA FROM RATINGS1FINAL_1_BACKEND WHERE RATINGS1FINAL_1_BACKEND.EXP_LIST = SUBQ_RATINGS1_PID.ROWID_MIMIR) < 5 OR CAST(SUBQ_RATINGS1_PID.RATINGS1_RATING AS REAL) IS NOT NULL AND CAST(SUBQ_RATINGS1_PID.RATINGS1_RATING AS REAL) < 5
```

**Comment by Legacy25** [comment]
I'm looking into this and it seems like it may be a percolater issue. Right now, the way the backend tables are being built are by running a query filtering out rows with non-null values in the column of interest. 

```
val rowidQuery =
      CTPercolator.propagateRowIDs(
        Select(IsNullExpression(Var(lens.schema()(cIndex)._1)), lens.source),
        true
      )
val rowidIterator = lens.db.query(rowidQuery)
```

Then the backend schema is generated by inspecting the first row of the result, which is necessary to know the type of the data column. For the simple demo, the rowidIterator returns 0 rows instead of the expected 1 row. Which means the backend table isn't created at all. 

[_Unrelated note_: I am assuming this is acceptable behavior when the case is indeed there are no rows with missing values. If the contents of the underlying tables change introducing missing values, necessitating the backend table, a subsequent call to createBackingStore will create the table.]

Returning to the issue, I think percolater is not producing the correct output for rowidQuery. The input to compile is -

```
SELECT[RATING IS NULL](
  PROJECT[ROWID_MIMIR <= ROWID_MIMIR, PID <= PID, RATING <= RATING, REVIEW_CT <= REVIEW_CT](
    PROJECT[ROWID_MIMIR <= ROWID_MIMIR, PID <= RATINGS1TYPED_PID, RATING <= RATINGS1TYPED_RATING, REVIEW_CT <= RATINGS1TYPED_REVIEW_CT](
      PROJECT[ROWID_MIMIR <= ROWID_MIMIR, RATINGS1TYPED_PID <= PID, RATINGS1TYPED_RATING <= RATING, RATINGS1TYPED_REVIEW_CT <= REVIEW_CT](
        PROJECT[ROWID_MIMIR <= ROWID_MIMIR, PID <= CAST(PID, {{ RATINGS1TYPED_0[] }}), RATING <= CAST(RATING, {{ RATINGS1TYPED_1[] }}), REVIEW_CT <= CAST(REVIEW_CT, {{ RATINGS1TYPED_2[] }})](
          PROJECT[ROWID_MIMIR <= ROWID_MIMIR, PID <= PID, RATING <= RATING, REVIEW_CT <= REVIEW_CT](
            PROJECT[ROWID_MIMIR <= ROWID_MIMIR, PID <= RATINGS1_PID, RATING <= RATINGS1_RATING, REVIEW_CT <= RATINGS1_REVIEW_CT](
              RATINGS1(RATINGS1_PID:string, RATINGS1_RATING:string, RATINGS1_REVIEW_CT:string // ROWID_MIMIR:rowid)
            )
          )
        )
      )
    )
  )
)
```

and the optimizedOper is -

```
PROJECT[ROWID_MIMIR <= ROWID_MIMIR, PID <= CAST(RATINGS1_PID, {{ RATINGS1TYPED_0[] }}), RATING <= CAST(RATINGS1_RATING, {{ RATINGS1TYPED_1[] }}), REVIEW_CT <= CAST(RATINGS1_REVIEW_CT, {{ RATINGS1TYPED_2[] }})](
  SELECT[CAST(RATINGS1_RATING, {{ RATINGS1TYPED_1[] }}) IS NULL](
    RATINGS1(RATINGS1_PID:string, RATINGS1_RATING:string, RATINGS1_REVIEW_CT:string // ROWID_MIMIR:rowid)
  )
)
```

Should not the Project over a non-deterministic Select be collapsed into a Project with ```MIMIR_CONDITION```?

Right now the iterator produced on this query is a ProjectionResultIterator over an NDInlineIterator over the basic deterministic iterator, which looks wrong to me...

**Comment by okennedy** [comment]
Why would it get collapsed.  It's getting evaluated inlined.  I think this might be the root of the problem though.  There's some canonical way to call a query where global VGTerms are properly removed in the compilation step that might not be getting used here.  

This stuff should probably get documented somewhere....

**Comment by okennedy** [comment]
Looks like you're going through db.compile(), which should correctly inline that select.  More digging required.

**Comment by okennedy** [comment]
Oy.  Looks like this behavior is a consequence of SQLite's CAST implementation silently obscuring errors.  
```
sqlite> SELECT CAST('shazbot' AS REAL);
0.0
```
Not sure what happened to bring this bug to the surface... but we should do something about it.  Maybe we go back to the old CAST operation as a non-deterministic operator...

**Comment by okennedy** [comment]
Fixed in #93
